File: app/models/provider_profile.py
from beanie import Document
from pydantic import BaseModel
from typing import Optional, List, Dict, Any
from datetime import datetime, timezone
from enum import Enum


class ProviderValidationStatus(str, Enum):
    pending = "pending"           # En attente d'examen
    reviewing = "reviewing"       # En cours d'examen
    waiting_docs = "waiting_docs" # En attente de documents complémentaires
    approved = "approved"         # Validé
    rejected = "rejected"         # Refusé


class Location(BaseModel):
    lat: float
    lng: float
    city: Optional[str] = None
    address: Optional[str] = None


class ValidationHistoryEntry(BaseModel):
    status: str
    note: Optional[str] = None
    admin_id: str
    at: datetime = datetime.now(timezone.utc)


class ProviderProfile(Document):
    user_id: str
    bio: Optional[str] = None
    skills: List[str] = []
    categories: List[str] = []
    location: Optional[Location] = None
    radius_km: float = 20.0
    portfolio: List[str] = []
    certificates: List[str] = []
    avg_rating: float = 0.0
    total_reviews: int = 0
    is_verified_provider: bool = False
    # --- Workflow de validation admin ---
    validation_status: ProviderValidationStatus = ProviderValidationStatus.pending
    validation_notes: Optional[str] = None
    id_card_url: Optional[str] = None       # pièce d'identité uploadée
    validation_history: List[ValidationHistoryEntry] = []
    updated_at: datetime = datetime.now(timezone.utc)

    class Settings:
        name = "provider_profiles"
        indexes = ["user_id", "validation_status", "is_verified_provider"]

File: service_matchmaking_mobile/lib/features/provider_profile/domain/entities/provider_profile.dart
import 'package:equatable/equatable.dart';

class ProviderProfileEntity extends Equatable {
  const ProviderProfileEntity({
    this.bio,
    this.skills = const [],
    this.categoryIds = const [],
    this.locationLat,
    this.locationLng,
    this.locationCity,
    this.locationAddress,
    this.radiusKm = 20,
    this.portfolio = const [],
    this.certificates = const [],
    this.avgRating = 0,
    this.totalReviews = 0,
    this.isVerifiedProvider = false,
  });

  final String? bio;
  final List<String> skills;
  final List<String> categoryIds;
  final double? locationLat;
  final double? locationLng;
  final String? locationCity;
  final String? locationAddress;
  final double radiusKm;
  final List<String> portfolio;
  final List<String> certificates;
  final double avgRating;
  final int totalReviews;
  final bool isVerifiedProvider;

  bool get hasLocation => locationLat != null && locationLng != null;

  ProviderProfileEntity copyWith({
    String? bio,
    List<String>? skills,
    List<String>? categoryIds,
    double? locationLat,
    double? locationLng,
    String? locationCity,
    String? locationAddress,
    double? radiusKm,
    List<String>? portfolio,
    List<String>? certificates,
  }) {
    return ProviderProfileEntity(
      bio: bio ?? this.bio,
      skills: skills ?? this.skills,
      categoryIds: categoryIds ?? this.categoryIds,
      locationLat: locationLat ?? this.locationLat,
      locationLng: locationLng ?? this.locationLng,
      locationCity: locationCity ?? this.locationCity,
      locationAddress: locationAddress ?? this.locationAddress,
      radiusKm: radiusKm ?? this.radiusKm,
      portfolio: portfolio ?? this.portfolio,
      certificates: certificates ?? this.certificates,
      avgRating: avgRating,
      totalReviews: totalReviews,
      isVerifiedProvider: isVerifiedProvider,
    );
  }

  @override
  List<Object?> get props => [
        bio,
        skills,
        categoryIds,
        locationLat,
        locationLng,
        locationCity,
        locationAddress,
        radiusKm,
        portfolio,
        certificates,
        avgRating,
        totalReviews,
        isVerifiedProvider,
      ];
}
